packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -74,14 +74,33 @@ export class GutenbergEditorPage {
 	 * @returns {Promise<Frame>} iframe holding the editor.
 	 */
 	async waitUntilLoaded(): Promise< Frame > {
-		await this.page.waitForLoadState( 'load' );
+		// `page.on` construct is used here instead of the more common `page.waitForResponse`.
+		// Using the latter causes this method to hang until the timeout is reached, since
+		// the `waitForResponse` method begins observing the network requests after the
+		// `nux?_envelope=1` request has been completed.
+		// On the other hand, `page.on` is able to monitor every request, including the
+		// one to the `nux` endpoint.
+		this.page.on( 'requestfinished', async ( request ) => {
+			const response = await request.response();
+			if ( ! response ) {
+				return;
+			}
+
+			if ( ! response.url().includes( 'nux?_envelope=1' ) ) {
+				return;
+			}
+
+			const body: { [ index: string ]: never } = await response.json();
+			if ( body[ 'body' ][ 'show_welcome_guide' ] === true ) {
+				await this.dismissWelcomeTourIfPresent();
+			}
```

Comment:
Let's use optional chaining here just to be extra safe:

```suggestion
			interface NuxPayload {
				body: {
					show_welcome_guide: boolean;
				};
			}
			const payload = ( await response.json() ) as NuxPayload;
			if ( payload?.body?.show_welcome_guide === true ) {
				await this.dismissWelcomeTour();
			}
```

Reply:
Interesting, I'll try this out.

Reply:
045a0ef365865b62ae5827960ff80cc17625aad0